Codecademy/gamut
# refactor(gamut-patterns): replacing incremental ids with auto-id

### Overview


* refactors incremental ids to use auto-id


### PR Checklist

- [x] Related to JIRA ticket: [web-2285]

#### Testing Instructions



* go to patterns in the test env
* the patterns should render
* go to a component that uses a `gamut-pattern` like `FloatingCard` 
* they should look identical to the way they do now

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/34613)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/1232)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
